src/pani/core.cljs

```diff
@@ -0,0 +1,161 @@
+(ns pani.core
+  (:refer-clojure :exclude [get-in set! reset! conj! swap! deref parents key])
+  (:require cljsjs.firebase
+            [clojure.string :as str]
+            [pani.registry :refer [register-listener]]))
+
+;; constants
+
+(def child-events
+  [:child-added
+   :child-changed
+   :child-moved
+   :child-removed])
+
+(def all-events
+  (conj child-events :value))
+
+;; utils
+
+;; FIXME: camel-case keys?
+;;        hydrate to custom vectors to preserve rich keys?
+;;        preserve sets (don't coerce to vector)
+;;        similarly preserve keywords as values
+
+(defn- hydrate [js-val]
+  (js->clj js-val :keywordize-keys true))
+
+(defn- serialize [clj-val]
+  (clj->js clj-val))
```

Comment:
Two reasons for this:

1. Yes exactly, I anticipate a bit more magic landing in the serialization, especially to provide better support for vectors, and possibly sets.

2. Moving to a shared namespace for JS and JVM, so useful to have one place to apply CLJX or reader conditionals.